sniproxy.cpp

```diff
@@ -190,7 +190,40 @@ SNIProxy::~SNIProxy()
 
 void SNIProxy::update()
 {
-    m_pixmap = QPixmap::fromImage(getImageNonComposite());
+    QImage newImage = getImageNonComposite();
+
+    if (m_pixmap.isNull()) {
+        m_pixmap = QPixmap::fromImage(newImage);
+    }
+    else if (!newImage.isNull()) {
+        auto isTransparentImage = [](const QImage &image) -> bool {
+            unsigned long long sum = 0;
+
+            for (int x = 0; x < image.width(); ++x) {
+                for (int y = 0; y < image.width(); ++y) {
+                    sum += qAlpha(image.pixel(x, y));
+                }
+            }
+
+            if (sum < 255) {
+                // All pixels more or less tranparent.
+                return true;
+            }
+
+            return false;
+        };
+
+        if (!isTransparentImage(newImage)) {
+            // Update icon only if it is at least partially opaque.
+            // This is just a workaround for X11 bug.
+            m_pixmap = QPixmap::fromImage(newImage);
+        }
+        else {
+            // Delete this before release.
```

Comment:
debug is fine, I'll just turn off all debug by default in the Q_LOGGING_CATEGORY line